### ISSUE TITLE
Kernel dies in quarto/integrals/surface_area.ipynb notebook

### ISSUE THREAD

**Issue by AronT-TLV** [jverzani/CalculusWithJuliaNotes.jl#113]
**OS version:** Mac OS X 13.4.1.(c)
**Hardware:** Mac M2 Max
**Julia Version** 1.9.1
**Conda version:** 23.5.2 (Conda installed into the `.Julia` directory by Conda package)
**Python version:** 3.10.12
**Error message:**
When running the command:
`quarto render quarto --to pdf --toc`
 the run fails with the following error:
```
[52/82] integrals/surface_area.qmd

Starting julia-1.9 kernel...Done


Executing 'surface_area.ipynb'
  Cell 1/30...Done
  Cell 2/30...Done
  Cell 3/30...Done
  Cell 4/30...Done
  Cell 5/30...Done
  Cell 6/30...Done
  Cell 7/30...Done
  Cell 8/30...Done
  Cell 9/30...Done
  Cell 10/30...Done
  Cell 11/30...Done
  Cell 12/30...Done
  Cell 13/30...2023-07-18 15:24:38,738 - traitlets - ERROR - Kernel died while waiting for execute reply.
```
**How to replicate the error running Julia in Jupyter or the CLI:**

If you copy the following code from the `quarto/integrals/surface.area.ipynb` notebook and run it in the REPL, or run just the cells in the notebook that contains this code, the code works without error:
```
using CalculusWithJulia
using Plots
using SymPy
using QuadGK
g(u) = 6 + 2sin(u)
f(u) = 2cos(u)
a, b = 0, 2pi
us = range(a, b, length=100)
plot(g.(us), f.(us), xlims=(-0.5, 9), aspect_ratio=:equal, legend=false)
plot!([0,0],[-3,3], color=:red, linewidth=5)    # y axis emphasis
plot!([3,9], [0,0], color=:green, linewidth=5)  # x axis emphasis
# lines of code in cell that crashes the server
S(u,v) = [g(u)*cos(v), g(u)*sin(v), f(u)] us = range(a, b, length=100)
vs = range(0, 2pi, length=100)
0.0:0.06346651825433926:6.283185307179586
ws = unzip(S.(us, vs'))   # reorganize data
surface(ws..., zlims=(-6,6), legend=false)

plot!([0,0], [0,0], [-3,3], color=:red, linewidth=5) # y axis emphasis
```
However, if you step through the Jupyter notebook from the very top, you get the "lost connection to kernel server" error, when you run the cell that contains the last few lines of code listed above.

Besides preventing building the pdf, this seems to indicate some underlying error in the `CalculuswithJulia` package. My suspicion is it is related to the `unzip` command, since the notation in the notebook seems to imply this is the first place you use it.

**Comment by jverzani** [comment]
Thanks for reporting back the issue. I'll have a look. (BTW, the pdf doesn't gracefully adjust for the plots and takes forever to build, so I haven't tested building it in awhile)

**Comment by jverzani** [comment]
Hmm, was hoping for an easy to fix typo. But am not finding it. I can compile with 1.8, but need to update the kernel I'm using with quarto.

**Comment by AronT-TLV** [comment]
FYI I tried building in `Quarto` after rebuilding the repository in both 1.8 and 1.7 (using JuliaUp) and rebuilding `iJulia` to deploy associated kernels. The `Quarto` build didn't work, although in 1.7 it just stopped without an error message. I also didn't do the above test until today, and that was only for 1.9. 

The PDF really isn't that important to me, I was using this as an opportunity to play around with `Quarto`. The web site is great, and that way I am always getting the latest version. The reason I reported the bug, is only because it seems like there is an underlying bug in `CalculuswithJulia` as the above test shows (since it was run outside `Quarto`) and that package I do want to use ;) I wasn't sure which was the best place to report the bug, but since the notebook was in this repository, I did it here.

**Comment by jverzani** [comment]
I tried building with a 1.9 kernel and it isn't working for me. Not sure why, but I have some vague memory that this was a known issue. Now I can't get Plots to work within Jupyter and 1.9, so there is something amiss, though not sure it is the CalculusWithJulia package, which doesn't do too much (though it certainly might be doing something incorrectly :)

**Comment by AronT-TLV** [comment]
Hmmm that is really strange. I got a very nice plot, when I ran the code in the REPL and in the Jupyter notebook if I just ran the four relevant boxes. I don't want to take too much of your time on this since I know how much work is involved in an open source project, and like I said the PDF output is not critical to me (and probably not for most people).  I am sure you have more important things that require your time.

If I run into any issues when going through the book that gives me more insights into an underlying problem I will let you know. In any case it's probably some subtle interactions amongst the millions of packages and  it will likely shake out in the future. I suggest we close this ticket :) In any case, many thanks for the great book and package!!!!!!

**Comment by jverzani** [comment]
No worries, it ins't the call to `unzip`, but the next call to `surface(...)` that is causing it to fail. I can repeat it. Now why it seems to work in the REPL and not the notebook is something I need to sort out, as the web page won't compile cleanly from scratch if not. So for now, I'll leave this open and hope to close soon.

**Comment by jverzani** [comment]
I tracked down the issue, which may be something other pages will have problems with. For that page, everything after the animated image doesn't show, whereas prior to that the calls to `surface` work fine. I've moved this particular animation to the end of the file. Hopefully there is somethign just about surface plots, as otherwise ...

Closed by #114
